markdown/assignments/sheet07/index.md

```diff
@@ -2,25 +2,43 @@
 type: assignment
 title: "Blatt 07: Logistische Regression"
 author: "Canan Yıldız (Türkisch-Deutsche Universität)"
-hidden: true
+hidden: false
 weight: 7
-sketch: true
+sketch: false
 ---
 
 
-Hier kommt der Inhalt für Blatt XYZ hin ... allgemeine einleitende Worte ...
+## A7.1: Logistische Regression (6P)
 
-## Aufgabe 1: XYZ (2P)
+*   Konstruieren Sie einen Datensatz $\mathcal{D}$ mit $m=100$ gleichförmig verteilten Zufallspunkten aus dem Bereich $\mathcal{X} = [−1, 1] \times [−1, 1] $.
+*   Wählen Sie auf ähnliche Weise zwei zufällige, gleichmäßig verteilte Punkte aus dem Bereich $ [−1, 1] \times [−1, 1] $. Verwenden Sie die Gerade, die durch diese zwei Punkte verläuft, als die Entscheidungsgrenze ihrer Zielfunktion $f$: Punkte auf der einen Seite der Linie sollen als +1 und die anderen als −1 klassifiziert werden. 
+*   Werten Sie Ihre Zielfunktion für jeden Datenpunkt $\mathbf{x}^{(j)}$ aus, um den entsprechenden Ausgangslabel $y^{(j)}$ zu erhalten.
 
-tbd
+Trainieren Sie ein Logistisches Regressionsmodell auf diesen Daten, um $h^{\*} = \sigma(w^T x)$ zu finden. Verwenden Sie dazu den Gradientenabstieg Algorithmus. Initialisieren Sie alle Gewichstwerte mit 0 und führen Sie 2000 Iterationen durch. Speichern Sie alle 100 Schritte die berechneten Kosten. Zeichnen Sie am Ende die Kosten als Diagramm über die Anzahl der Iterationen auf.
+
+Experimentieren Sie mit unterschiedlichen Lernraten, z.B.   = 0.1,   = 0.01,   = 0.001 und vergleichen Sie die Kosten-Diagramme.
+Sie können das folgende <a href="files/Logistische_Regression_Starter.ipynb" target="_blank" rel="nofollow noopener noreferrer"><strong>Jupyter Notebook</strong></a> als Startpunkt benutzen. Sie können alternativ auch eine andere Programmiersprache und/oder einen anderen Datensatz (z.B. zufällig generierter Datensatz mittels Numpy and Scikit-Learn) verwenden.
+
+*Thema*: Verständnis und Implementierung Logistische Regression
+
+
+
+## A7.2: Merkmal Skalierung (4P)
+
+Abbildung 1 und Abbildung 2 zeigen die [Höhenlinien](https://de.wikipedia.org/wiki/H%C3%B6henlinie) ([Contour Lines](https://en.wikipedia.org/wiki/Contour_line)) von zwei Kostenfunktionen [Quelle](https://academo.org/demos/contour-plot/).
```

Comment:
Hier fehlt noch eine Lizenz-Angabe.

Wenn die Bilder bei uns im Repo liegen sollen (was sie aktuell tun), müssten wir es irgendwie hinkriegen, hier eine Annotation mit Quelle und Lizenz an den einzelnen Bilddateien herzustellen. Das könnten evtl. gleichnamige Textdateien sein mit Herkunft und Lizenz als Inhalt.

Besser wäre es aus meiner Sicht, die Bilder per Link nur in der Webseite einzubinden. Dann sind sie nicht Bestandteil des Repos und in der Webseite hätten wir die nötige Quellenangabe. => `![Abbildung1](URL-zum-Bild)`.

Reply:
Oder, da die Bilder eigentlich trivial sind: Schnell selbst nachmalen mit Excel o.ä.?

Reply:
Ich hatte im Text die Quelle zu der Webseite angegeben (ist alles frei nutzbar).
Das sieht man aber nicht, ich selber hatte es vergessen und eben zufällig wieder bemerkt. 
Diese Angabe sollte eigentlich unter den Bildern stehen.

Ich habe die Bilder aber gleich selbst gemalt, ist einfacher und sauberer so.
Danke für den Hinweis!